src/unix/mod.rs

```diff
@@ -206,9 +206,19 @@ pub const PRIO_MAX: ::c_int = 20;
 cfg_if! {
     if #[cfg(dox)] {
         // on dox builds don't pull in anything
-    } else if #[cfg(all(not(stdbuild), feature = "use_std"))] {
+    } else if #[cfg(all(not(stdbuild),
+                        feature = "use_std",
+                        not(any(target_os = "macos",
+                                target_os = "ios")
+                        )))] {
         // cargo build, don't pull in anything extra as the libstd  dep
         // already pulls in all libs.
+    } else if #[cfg(all(not(stdbuild), feature = "use_std"))] {
+        // except on macOS and iOS, where we must link with lib resolv
+        // for res_init, despite libsystem_info including it:
+        // http://blog.achernya.com/2013/03/os-x-has-silly-libsystem.html
+        #[link(name = "resolv")]
```

Comment:
Let's start off by disabling linking to `resolv` everywhere. Let's then only enable it in the `not(stdbuild)` blocks to ensure we don't accidentally break crates.io. By only enabling it in `not(stdbuild)` then we can't test it in this PR, but it's ok to leave it enabled on all platforms you have it defined on currently.